lib/elastomer/client/scroller.rb

```diff
@@ -128,19 +135,14 @@ class Scroller
       #
       def initialize( client, query, opts = {} )
         @client = client
-        @query  = query
 
-        @index       = opts.fetch(:index, nil)
-        @type        = opts.fetch(:type, nil)
-        @scroll      = opts.fetch(:scroll, '5m')
-        @size        = opts.fetch(:size, 50)
-        @search_type = opts.fetch(:search_type, nil)
+        @opts = DEFAULT_OPTS.merge({ :body => query }).merge(opts)
 
         @scroll_id = nil
         @offset = 0
       end
 
-      attr_reader :client, :query, :index, :type, :scroll, :size, :search_type, :scroll_id
+      attr_reader :client, :query, :scroll_id
```

Comment:
Good catch - I forgot that they're visible outside of the class. I checked and I don't see any references to them elsewhere in the library. If it's likely that this is a user-facing breaking change, they could be implemented by reading from the `@opts` hash, or by putting `#scroll_opts` back. I'll defer to your judgement here.